Спринт 15

justannnnnn opened the pull request:
Refactoring code for Clean Architecture

justannnnnn committed:
Refactoring code for Clean Architecture

sukhoikms27 commented on app/src/main/java/com/example/playlistmaker/presentation/theme_manager/ThemeManager.kt:
```diff
@@ -0,0 +1,25 @@
+import com.example.playlistmaker.Creator
+import com.example.playlistmaker.domain.api.ThemeRepository
+
+object ThemeManager {
```
работу с темой так же нужно разделить на слои + вынести репозиторий в конструктор

sukhoikms27 commented on app/src/main/java/com/example/playlistmaker/presentation/theme_manager/SwitchTheme.kt:
```diff
@@ -0,0 +1,12 @@
+import android.app.Application
+import com.example.playlistmaker.Creator
+
+class SwitchTheme: Application() {
```
можно было оставить файл `App` наследуемый от `application` - он нам в дальнейшем еще пригодиться

sukhoikms27 commented on app/src/main/java/com/example/playlistmaker/data/SearchHistoryRepositoryImpl.kt:
```diff
@@ -0,0 +1,42 @@
+import com.example.playlistmaker.domain.models.Track
+import com.google.gson.Gson
+
+class SearchHistoryRepositoryImpl(private val sp: SharedPreferences): SearchHistoryRepository {
```
Рекомендация:
`sp` - такие сокращения не всегда удобны, лучше назвать чуть подробнее, например `prefs`

sukhoikms27 commented on app/src/main/java/com/example/playlistmaker/data/SharedPreferencesRepositoryImpl.kt:
```diff
@@ -0,0 +1,20 @@
+import android.content.SharedPreferences
+import com.example.playlistmaker.domain.api.ThemeRepository
+
+class SharedPreferencesRepositoryImpl(private val sp: SharedPreferences): ThemeRepository {
```
Рекомендация:
`sp` - такие сокращения не всегда удобны, лучше назвать чуть подробнее, например prefs

sukhoikms27 replied:
Рекомендация:
`SharedPreferencesRepositoryImpl` - здесь название слишком абстрактное, к тому же в соседнем репозитории мы тоже работает с префсами, но название соответствует фиче, поэтому рекомендую навать ближе к контексту класса: `ThemeRepositoryImpl`

sukhoikms27 commented on app/src/main/java/com/example/playlistmaker/domain/impl/TrackInteractorImpl.kt:
```diff
@@ -0,0 +1,26 @@
+        executor.execute {
+            try {
+                val tracks = repository.searchTracks(term)
+                if (tracks!!.isEmpty()) {
```
Рекомендация:
лучше `tracks.isNullOrEmpty()`, тогда мы избавимся от `!!`

sukhoikms27 commented on app/src/main/java/com/example/playlistmaker/presentation/ui/maker/TrackAdapter.kt:
```diff
@@ -0,0 +1,33 @@
+package com.example.playlistmaker.presentation.ui.maker
```
лучше положить в пакет `search`

sukhoikms27 commented on app/src/main/java/com/example/playlistmaker/presentation/ui/search/SearchActivity.kt:
```diff
@@ -21,103 +21,104 @@ import androidx.appcompat.app.AppCompatActivity
         }
 
+        interactor = Creator.provideTrackInteractor()
+        searchHistoryRepository = Creator.provideSearchHistoryRepository()
```
нет прослойки интерактора для истории поиска

justannnnnn committed:
Fixed bugs from review:
 - added ThemeInteractor and ThemeRepository
 - moved TrackAdapter and TrackViewHolder to search folder
 - added SearchHistoryInteractor
